ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -663,12 +674,15 @@ def establish_trust_with_ad(master, ad_domain, extra_args=()):
     master.run_command(['klist'])
     master.run_command(['smbcontrol', 'all', 'debug', '100'])
 
-    run_repeatedly(master,
-                   ['ipa', 'trust-add',
-                    '--type', 'ad', ad_domain,
-                    '--admin', 'Administrator',
-                    '--password'] + list(extra_args),
-                   stdin_text=master.config.ad_admin_password)
+    if shared_secret:
+        extra_args += ['--trust-secret']
+        stdin_text = shared_secret
+    else:
+        extra_args += ['--admin', 'Administrator', '--password']
```

Comment:
should not we leave this as default option like earlier in run_repeatedly as this is not extra_args, its mandatory part during trust-add.
we can just add extra_args for --trust-secret and stdin_text in if section and else section can have just stdin_text for password.

Reply:
shared secret should not be used together with --admin option